api/grpcserver/interface.go

```diff
@@ -86,3 +86,8 @@ type meshAPI interface {
 type oracle interface {
 	ActiveSet(context.Context, types.EpochID) ([]types.ATXID, error)
 }
+
+type postConnectionListener interface {
+	Connected(client activation.PostClient)
+	Disconnected(client activation.PostClient)
+}
```

Comment:
Note: the system is more interested in post-services _registering_ with an ID than just _connecting_. (I know that it doesn't send the ID _yet_ - TBD).

What's the purpose of `Disconnected()`?

Reply:
I used a pub-sub like pattern. Any component of the node that is interested in post services connecting or disconnecting might use this interface to be informed about a service opening a stream to the node (or dropping the stream).

Since a component is informed when a service opened a stream ("Connected") it is also informed when this service closes the stream (the object passed into the service will be the same object in both cases / `==` will be true) and can free resources when this happens.